ichnaea/tests/test_rate_limit.py

```diff
@@ -0,0 +1,57 @@
+import time
+
+from ichnaea.service.base import rate_limit
+from ichnaea.tests.base import (
+    RedisIsolation,
+    TestCase,
+)
+
+
+class TestLimiter(TestCase, RedisIsolation):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestLimiter, cls).setup_redis()
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestLimiter, cls).teardown_redis()
+
+    def tearDown(self):
+        self.cleanup_redis()
+
+    def test_limiter_maxrequests(self):
+        api_key = 'key_a'
+        maxreq = 5
+        expire = 1
+        for i in range(maxreq):
+            self.assertFalse(rate_limit(
+                self.redis_client,
+                api_key,
+                maxreq=maxreq,
+                expire=expire,
+            ))
+        self.assertTrue(rate_limit(
+            self.redis_client,
+            api_key,
+            maxreq=maxreq,
+            expire=expire,
+        ))
```

Comment:
I know this is just a moved existing test, but if you feel like it, https://github.com/mozilla/ichnaea/pull/432/files#r29757957 applies here as well (second rollover right before the assertTrue call).